diff -uraN firewalld-2.4.3/config/firewalld.service.in omv-firewalld-2.4.3/config/firewalld.service.in
--- firewalld-2.4.3/config/firewalld.service.in	2026-06-24 19:51:15.000000000 +0200
+++ omv-firewalld-2.4.3/config/firewalld.service.in	2026-07-25 22:02:52.922394214 +0200
@@ -18,7 +18,7 @@
 DevicePolicy=closed
 KeyringMode=private
 LockPersonality=yes
-MemoryDenyWriteExecute=yes
+MemoryDenyWriteExecute=no
 PrivateDevices=yes
 ProtectClock=yes
 ProtectControlGroups=yes
